packages/cli-repl/src/mongosh-repl.ts

```diff
@@ -228,8 +229,22 @@ class MongoshNodeRepl {
   async eval(originalEval: asyncRepl.OriginalEvalFunction, input: string, context: any, filename: string): Promise<any> {
     this.lineByLineInput.enableBlockOnNewLine();
     const shellEvaluator = this.runtimeState().shellEvaluator;
+
     try {
-      return await shellEvaluator.customEval(originalEval, input, context, filename);
+      let error: Error | undefined = undefined;
+      let result: any = undefined;
+      try {
+        result = await shellEvaluator.customEval(originalEval, input, context, filename);
+      } catch (e) {
+        error = e;
+      }
+
+      await this.updatePrompt();
```

Comment:
We could also move this into the `finally`, right?

Reply:
Yep, true... will move it.